src/sync/PostSyncProcessor.js

```diff
@@ -0,0 +1,157 @@
+/**
+ * All utility functions for processing synced records after being synced. These should changes
+ * should be detected by SyncQueue and sent back to the server on next sync.
+ */
+
+import { getNextNumber, NUMBER_SEQUENCE_KEYS } from '../database/utilities';
+
+const { REQUISITION_SERIAL_NUMBER, SUPPLIER_INVOICE_NUMBER } = NUMBER_SEQUENCE_KEYS;
+
+export class PostSyncProcessor {
+  constructor(database) {
+    this.database = database;
+    this.recordQueue = new Map(); // Map of [recordId, recordType]
```

Comment:
Interesting, the first time Map() is used in mobile, I guess it's easier to iterate